src/rgw/rgw_auth.h

```diff
@@ -101,6 +101,9 @@ class Identity {
 
   /* write any auth-specific fields that are safe to expose in the ops log */
   virtual void write_ops_log_entry(rgw_log_entry& entry) const {};
+
+  /* Returns true if the identity is acting on behalf of another user */
+  virtual bool is_impersonating() const { return false; }
```

Comment:
just trying to think of better names for `is_impersonating()` and `is_wrapped`

all that `process_request()` and `RGWOp`s really need from this `Identity` abstraction is to know whether or not permission checks should be enforced

it's probably simplest for `SysReqApplier` to just override `is_admin_of()` to accomplish this part: return true for system users without `rgwx-perm-check-uid`, otherwise forward to `DecoratedApplier<T>::is_admin_of()`. because `LocalApplier::is_admin_of()` returns `user_info.admin || user_info.system`, auth bypass will work for both system-wrapping-admin and system-wrapping-system cases

then `s->system_request` would only be used to enable multisite's API extensions

Reply:
make sense. but i still had to hold a bool in `SysReqApplier` so we can know whether this is an impersonated request or not for both `load_acct_info()` and `is_admin_of()` override.

i also dropped the input for `is_admin_of()` and changed it to `is_admin()` for better readability.